connector.go

```diff
@@ -5,48 +5,114 @@ import "time"
+	// use a DeleteRequest object + Do() instead.
 	DeleteTyped(space, index interface{}, key interface{}, result interface{}) (err error)
+	// Deprecated: the method will be removed in the next major version,
+	// use a UpdateRequest object + Do() instead.
 	UpdateTyped(space, index interface{}, key, ops interface{}, result interface{}) (err error)
+	// Deprecated: the method will be removed in the next major version,
+	// use a CallRequest object + Do() instead.
 	CallTyped(functionName string, args interface{}, result interface{}) (err error)
+	// Deprecated: the method will be removed in the next major version,
+	// use a Call16Request object + Do() instead.
 	Call16Typed(functionName string, args interface{}, result interface{}) (err error)
+	// Deprecated: the method will be removed in the next major version,
+	// use a Call17Request object + Do() instead.
 	Call17Typed(functionName string, args interface{}, result interface{}) (err error)
+	// Deprecated: the method will be removed in the next major version,
+	// use an EvalRequest object + Do() instead.
 	EvalTyped(expr string, args interface{}, result interface{}) (err error)
+	// Deprecated: the method will be removed in the next major version,
+	// use an ExecuteRequest object + Do() instead.
 	ExecuteTyped(expr string, args interface{}, result interface{}) (SQLInfo, []ColumnMetaData, error)
 
+	// Deprecated: the method will be removed in the next major version,
+	// use a SelectRequest object + Do() instead.
 	SelectAsync(space, index interface{}, offset, limit uint32, iterator Iter, key interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use an InsertRequest object + Do() instead.
 	InsertAsync(space interface{}, tuple interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use a ReplaceRequest object + Do() instead.
 	ReplaceAsync(space interface{}, tuple interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use a DeleteRequest object + Do() instead.
 	DeleteAsync(space, index interface{}, key interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use a UpdateRequest object + Do() instead.
 	UpdateAsync(space, index interface{}, key, ops interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use a UpsertRequest object + Do() instead.
 	UpsertAsync(space interface{}, tuple interface{}, ops interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use a CallRequest object + Do() instead.
 	CallAsync(functionName string, args interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
+	// use a Call16Request object + Do() instead.
 	Call16Async(functionName string, args interface{}) *Future
+	// Deprecated: the method will be removed in the next major version,
```

Comment:
Is the common way for Go to deprecate is through documentation? It seems that emitting a warning is rather complicated.

Reply:
Yes, it is: https://go.dev/blog/godoc

`Sometimes a struct field, function, type, or even a whole package becomes redundant or unnecessary, but must be kept for compatibility with existing programs. To signal that an identifier should not be used, add a paragraph to its doc comment that begins with “Deprecated:” followed by some information about the deprecation.`

Some tools support it (include our linter), see https://stackoverflow.com/questions/7849663/how-do-you-mark-code-as-deprecated-in-go .